pkg/apis/core/v1beta1/helper/common.go

```diff
@@ -100,21 +100,23 @@ func HasOperationAnnotation(annotations map[string]string) bool {
 		annotations[v1beta1constants.GardenerOperation] == v1beta1constants.GardenerOperationMigrate
 }
 
-// ResourceReferencesEqual returns true when at least one of the Secret/ConfigMap resource references has changed.
+// ResourceReferencesEqual returns true when none of the Secret/ConfigMap/WorkloadIdentity resource reference has changed.
 func ResourceReferencesEqual(oldResources, newResources []gardencorev1beta1.NamedResourceReference) bool {
 	var (
 		oldNames = sets.New[string]()
 		newNames = sets.New[string]()
 	)
 
 	for _, resource := range oldResources {
-		if resource.ResourceRef.APIVersion == "v1" && sets.New("Secret", "ConfigMap").Has(resource.ResourceRef.Kind) {
+		if resource.ResourceRef.APIVersion == "v1" && sets.New("Secret", "ConfigMap").Has(resource.ResourceRef.Kind) ||
+			resource.ResourceRef.APIVersion == "security.gardener.cloud/v1alpha1" && resource.ResourceRef.Kind == "WorkloadIdentity" {
```

Comment:
This can be extracted to a function and reused for the `newResources` as well:

```suggestion

	validResourceRef = func(resource) bool {
		return resource.ResourceRef.APIVersion == "v1" && sets.New("Secret","ConfigMap").Has(resource.ResourceRef.Kind) || resource.ResourceRef.APIVersion == "security.gardener.cloud/v1alpha1" && resource.ResourceRef.Kind == "WorkloadIdentity" 
	}

	for _, resource := range oldResources {
		if validResourceRef(resource)
```

We could also use:
- `corev1.SchemeGroupVersion.String()`
- `securityv1alpha1.SchemeGroupVersion.String()`